File: test/fakers/lead_faker.dart
import 'package:faker/faker.dart';
import 'package:sertton/core/marketing/dtos/lead_dto.dart';

class LeadFaker {
  static final _faker = Faker();

  static LeadDto fakeDto({String? email, String? name}) {
    return LeadDto(
      email: email ?? _faker.internet.email(),
      name: name ?? _faker.person.name(),
    );
  }

  static List<LeadDto> fakeManyDto({
    int count = 10,
    String? email,
    String? name,
  }) {
    return List.generate(count, (index) => fakeDto(email: email, name: name));
  }
}

File: src/core/marketing/entities/fakers/lead-faker.ts
import { faker } from "@faker-js/faker"

import type { Lead } from "../lead"

export class LeadFaker {
  static fake(lead?: Partial<Lead>): Lead {
    return { email: faker.internet.email(), name: faker.person.fullName(), ...lead }
  }

  static fakeMany(count = 10, lead?: Partial<Lead>): Lead[] {
    return Array.from({ length: count }, () => LeadFaker.fake(lead))
  }
}
